src/index.ts

```diff
@@ -122,12 +123,20 @@ export function datadog<TEvent, TResult>(
     }
 
     let result: TResult | undefined;
+    let localResult: any | undefined;
     let error: any;
     let didThrow = false;
 
     try {
-      result = await traceListener.onWrap((localEvent: TEvent, localContext: Context) => {
-        return promHandler(localEvent, localContext);
+      result = await traceListener.onWrap(async (localEvent: TEvent, localContext: Context) => {
+        try {
+          localResult = await promHandler(localEvent, localContext);
```

Comment:
This feels like a significant change to me, should this be done within the `promisifiedHandler`?